src/shared/tpm2-util.c

```diff
@@ -1683,9 +1904,23 @@ int tpm2_unseal(const char *device,
         if (r < 0)
                 return r;
 
-        r = tpm2_make_primary(c.esys_context, &primary, primary_alg, NULL);
-        if (r < 0)
-                return r;
+        /* If their is a primary key we trust, like an SRK, use it */
+        if (primary_buf) {
+                log_debug("Found existing SRK key to use, deserializing ESYS_TR");
+                rc = sym_Esys_TR_Deserialize(
+                                c.esys_context,
+                                primary_buf,
+                                primary_buf_size,
+                                &primary);
+                if (rc != TSS2_RC_SUCCESS)
+                        return log_error_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE),
+                                               "Failed to unmarshal primary key: %s", sym_Tss2_RC_Decode(rc));
+        /* old callers without an SRK still need to create a key */
+        } else {
+                r = tpm2_make_primary(c.esys_context, &primary, primary_alg, false, NULL);
```

Comment:
maybe tpm2_make_primary() should now be renamed to `tpm2_make_transient_primary()` to make clear what this is about